percival/carrier/configuration.py

```diff
@@ -667,35 +666,56 @@ class SetpointGroupParameters(object):
     """
     Loads groups of controls description from an INI file.
     """
-    def __init__(self, ini_file):
+    def __init__(self):
         self.log = logging.getLogger(".".join([__name__, self.__class__.__name__]))
-        self._ini_filename = None
-        self._ini_buffer = None
-        try:
-            self._ini_filename = find_file(ini_file)
-        except:
-            # If we catch any kind of exception here then treat the parameter as the configuration
-            self._ini_buffer = StringIO(str(ini_file))
 
-    def load_ini(self):
+        self.conf = ConfigParser(dict_type=OrderedDict)
+        self.conf.optionxform = str; # this makes the option names case-sensitive
+
+        self.setpoint2section = {};
+
+    def load_ini(self, ini_file):
         """
         Loads and parses the data from INI file. The data is stored internally in the object and can be retrieved
-        through the property methods
+        through the property methods. Data from a previous load is kept, or silently overwritten.
         """
-        self.conf = SafeConfigParser(dict_type=OrderedDict)
-        self.conf.optionxform = str
-        if self._ini_filename:
-            self.conf.read(self._ini_filename)
-            self.log.info("Read Setpoint Groups INI file: %s", self._ini_filename)
+
+        _ini_filename = None
+        _ini_buffer = None
+
+        try:
+            _ini_filename = find_file(ini_file)
+        except:
+            # If we catch any kind of exception here then treat the parameter as the configuration
+            _ini_buffer = ini_file;
+
+        if _ini_filename:
+            self.conf.read(_ini_filename)
+            self.log.debug("Read Setpoint Groups INI file: %s", _ini_filename)
         else:
-            self.conf.readfp(self._ini_buffer)
-            self.log.info("Read Setpoint Groups INI object %s", self._ini_buffer)
+            self.conf.read_string(_ini_buffer)
+            self.log.debug("Read Setpoint Groups INI string %s", _ini_buffer)
```

Comment:
Are you sure you only want this (and the other messages) as debug?  In the case of logging to a logserver (graylog) instead of to terminal when using a production system, if a failure occurred you would always be able to see this message showing the setpoints (and other configurations), which could help to explain why the system failed.

Reply:
info is for short log messages. Before the change it didn't print the config-string, just an IO object description so it has improved. Also note that it log.info the section-names which includes the setpoint names.